### ISSUE TITLE
The Load Movies dialog is slow

### ISSUE THREAD

**Issue by RetroEdit** [TASVideos/BizHawk#2198]
@alyosha-tas brought this up in #2197. There are actually a few reasons for this. One main reason is that loading movies for previewing currently requires loading the entire movie file so the frame count can be displayed. There are two potential solutions here: add a frame count field to the header so the movie's frame count can be previewed without excessive loading. Or just don't display the frame count in the preview.

Besides the basic slowness of loading entire movies just for the purpose of previewing them, I think there's some double-loading going on for some reason. Each movie appears to be loaded twice from some preliminary debugging.

**Comment by adelikat** [comment]
Fix the double loading, don't add a frame count.  

I very much oppose the frame count idea due to having to keep up with it on every frame change (or it be out of sync by assuming current frame = framecount, see cyclecount problems).

Also, I have big concerns about someone misusing it to be an accurate frame count when the input log count is the source of truth

**Comment by RetroEdit** [comment]
Couldn't we just deal with it upon serialization: get the length of the input log when serializing the movie file and that way it will be definite.

Also, it doesn't have to be exposed to the general interface code; it could just be specific to preloading by design.

The reason I would really like to see this happen is because loading entire movies just for the purpose of getting the frame count is extremely wasteful. I understand the rationale of not wanting the field to be misused, but wasting time on previewing movie information is extremely annoying.

**Comment by adelikat** [comment]
That's true, it's completely calculable without a core, so it could be at least consistently accurate.  Meh, I guess I support this.  If anyone starts using it in the tasvideos parser, I will just scold them appropriately

**Comment by adelikat** [comment]
I've addressed the unnecessary loading that was a series of regressions from the original intent.  Now it only loads the header and input log when pre-loading, and the necessary loading of the file twice was eliminated.

This is the extent of what I wanted to accomplish for the 2.5 milestone, so I'm removed the tag.

Leaving this ticket open to track the change to have frame count in the header.  In pre-loading, if this value is present, it should skip loading of the input log, but if it is not there, it should continue to load it.
